src/main/java/org/humancellatlas/ingest/submission/web/SubmissionLinkMapController.java

```diff
@@ -0,0 +1,90 @@
+package org.humancellatlas.ingest.submission.web;
+
+import lombok.Getter;
+import org.humancellatlas.ingest.biomaterial.Biomaterial;
+import org.humancellatlas.ingest.biomaterial.BiomaterialRepository;
+import org.humancellatlas.ingest.file.File;
+import org.humancellatlas.ingest.file.FileRepository;
+import org.humancellatlas.ingest.process.Process;
+import org.humancellatlas.ingest.process.ProcessRepository;
+import org.humancellatlas.ingest.protocol.ProtocolRepository;
+import org.humancellatlas.ingest.submission.SubmissionEnvelope;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@RestController
+public class SubmissionLinkMapController {
+
+    @Autowired
+    BiomaterialRepository biomaterialRepository;
+    @Autowired
+    FileRepository fileRepository;
+    @Autowired
+    ProcessRepository processRepository;
+    @Autowired
+    ProtocolRepository protocolRepository;
+
+
+    @RequestMapping(path = "/submissionEnvelopes/{sub_id}/linkingMap", method = RequestMethod.GET)
+    @ResponseBody
+    public SubmissionLinkingMap getSubmissionLinkMap(@PathVariable("sub_id") SubmissionEnvelope submissionEnvelope) {
+        return new SubmissionLinkingMap(submissionEnvelope);
+    }
+
+    @Getter
+    public class SubmissionLinkingMap {
+        final Dictionary<String, ProcessLinkingMap> processes = new Hashtable<>();
+        final Dictionary<String, UUID> protocols = new Hashtable<>();
```

Comment:
I think this line is unused. Protocols are now returned as part of `this.processes`